build.project

```diff
@@ -30,21 +23,39 @@ if [ -z "$TARGET" ]; then
     exit 1
 fi
 
+
 TARGET_FAMILY=$(echo "$TARGET" | awk -F- '{ print $1 }')
-TARGET_SUBFAMILY="$(echo "$TARGET" | awk -F- '{ print $1"-"$2 }')"
+TARGET_SUBFAMILY="$(echo "$TARGET" | awk -F- '{ print $2 }')"
+TARGET_PROJECT="$(echo "$TARGET" | awk -F- '{ print $3 }')"
+
+
+BUILD_TARGET_FILE="_targets/build.project.${TARGET_FAMILY}-${TARGET_SUBFAMILY}"
+BUILD_PROJECT_FILE="_projects/build.project.${TARGET_FAMILY}-${TARGET_SUBFAMILY}-${TARGET_PROJECT}"
+
 
-BUILD_TARGET_FILE="_targets/build.project.${TARGET_SUBFAMILY}"
 if [ ! -f "$BUILD_TARGET_FILE" ]; then
     b_list_available_targets "TARGET='$TARGET' is unavailable for this project"
     exit 1
 fi
 
 . "$BUILD_TARGET_FILE"
 
+
+if [ -f "$BUILD_PROJECT_FILE" ]; then
+	. "$BUILD_PROJECT_FILE"
+else
+	if [ ! -z "$TARGET_PROJECT" ] || [ "${TARGET: -1}" == "-" ]; then
```

Comment:
it's always better to use "=" instead of "==" for string comparison in sh ("==" is bash extension, "=" is POSIX sh compliant)